### ISSUE TITLE
$agent stops responding after minimize

### ISSUE THREAD

**Issue by lixas** [jasonelle-archive/jasonelle-v2#33]
I'm trying new concept to use agent for data storage together with TaffyDB (javascript based library). Main concept works and i'm able to create and manipulate data. 

**Describe the bug**
But I have encountered issue, that $agent stops responding if I minimize app and recall from background later (even after 3 seconds). I have to close it and reopen to fully use $agent functionality

**Example Jasonette Document URL**
http://www.ignas.in/android/taffy.json
http://www.ignas.in/android/taffy.html

**To Reproduce**
1. Open application
2. Click all buttons and ensure everything is working
3. Send app to background by going to your android home
4. Restore application
5. NONE of buttons works that previously worked

**Expected behavior**
$agent continue to work as before minimize

**Smartphone (please complete the following information):**
 - Device: Xiaomi Mi4 Note
 - OS: Android 7

Any ideas whats going on?

**Comment by clsource** [comment]
I'm getting an error on the parser

![imagen](https://user-images.githubusercontent.com/292738/56517442-f1a96680-650a-11e9-94f8-ff405075c6af.png)

This is the raw data

```json

{
  "$jason": {
    "head": {
      "title": "$agent.request Demo",
      "description": "Jasonette to Agent request demo",
      "agents": {
        "crypto": {
          "url": "http://www.ignas.in/android/taffy.html"
        }
      },
      "actions": {
        "$load": {
          "type": "$set",
          "options": {
            "items": []
          },
          "success": {
            "type": "$render"
          }
        },
        "add": {
          "type": "$set",
          "options": {
            "items": "{{[$jason.result].concat($get.items)}}"
          },
          "success": {
            "type": "$render"
          }
        },
// ----------------------------------------------------------------
        "dbinsert": {
          "type": "$agent.request",
          "options": {
            "id": "crypto",
            "method": "addOne",
            "params": ["{{$get.firstname}}", "{{$get.lastname}}"]
          },
          "success": {
            "trigger": "add",
            "options": {
              "result": "{{$jason}}"
            }
          }
        },
        "showdb": {
          "type": "$agent.request",
          "options": {
            "id": "crypto",
            "method": "showAll"
          },
          "success": {
            "trigger": "add",
            "options": {
              "result": "{{$jason}}"
            }
          }
        },
        "persist":{
          "type": "$agent.request",
          "options": {
            "id": "crypto",
            "method": "store"
          },
          "success": {
            "trigger": "add",
            "options": {
              "result": "{{$jason}}"
            }
          }
        },
        "retrieve":{
          "type": "$agent.request",
          "options": {
            "id": "crypto",
            "method": "retrieve"
          },
          "success": {
            "trigger": "add",
            "options": {
              "result": "{{$jason}}"
            }
          }
        }
// ----------------------------------------------------------------
      },
      "templates": {
        "body": {
          "header": {
            "title": "jasonette Taffy demo"
          },
          "sections": [{
            "items": {
              "{{#concat}}": [
                [

                  {
                  "type": "horizontal",
                  "components": [{
                    "type": "textfield",
                    "name": "firstname",
                    "placeholder": "Enter first name"
                  }, {
                      "type": "switch",
                      "name": "jsn",
                      "value": "false"
                  }]
                    
                  },
                  
                  {
                    "type": "textfield",
                    "name": "lastname",
                    "placeholder": "LastName"
                  },
//                  {
//                      "type": "switch",
//                      "name": "jsn",
//                      "value": "false"
//                  },
// --------------------------------------------------------------------
                  {
                    "type": "button",
                    "text": "Insert record",
                    "action": {
                      "trigger": "dbinsert"
                    }
                  },
                  {
                    "type": "button",
                    "text": "Show All Data",
                    "action": {
                      "trigger": "showdb",
                      "options": {
                        "sync": true
                      }
                    }
                  },
                  {
                    "type": "horizontal",
                    "components": [{
                      "type": "button",
                      "text": "Read persist",
                      "action": {
                        "trigger": "retrieve"
                      }
                    }, {
                      "type": "button",
                      "text": "Write to stotage",
                      "action": {
                        "trigger": "persist"
                      }
                    }]
                }
// --------------------------------------------------------------------
                ],
                {
                  "{{#each $get.items}}": {
                    "type": "label",
                    "text": "{{this}}"
                  }
                }
              ]
            }
          }]
        }
      }
    }
  }
}
```

**Comment by lixas** [comment]
You may remove "comments" (because it is not valid JSON) but it is not the case in here. Even original agent samples are behaving the same for me...

**Comment by clsource** [comment]
Ok then this needs futher investigation

**Comment by lixas** [comment]
I have updated my taffy.json so comment wouldn't invalidate JSON document, but it should not be the case. I have tried crypto $agent examples ( https://github.com/jasonelle/jasonette-agentjason/tree/master/request/jasonette2agent ) and they do behave the same. I don't know how could i investigate further, because have no clue how to debug this. Maybe, at least, someone could confirm this behaviour on android?
